pkg/cmd/roachtest/tests/follower_reads.go

```diff
@@ -765,6 +765,12 @@ func verifySQLLatency(
 // follower reads) for all but a few nodes (toleratedNodes). We tolerate a few
 // wacky quantas because, around leaseholder changes, the one node's ratio drops
 // to zero (the new leaseholder), and another one's rises (the old leaseholder).
+//
+// TODO(darrylwong): This check is broken for multi-tenant. `follower_reads.success_count`
+// is a storage level metric while `sql.select.count` is application level metric but both
+// are queried through the system tenant. On shared-process mode, this means `sql.select.count`
+// returns the number of selects across all tenants, not just the tenant the test is run on.
+// On separate-process mode, `sql.select.count` returns only the system tenant's selects.
```

Comment:
Hmm, still confirming this but it actually sounds like separate process tenants/sql pods don't support TSDB writes which would make this test incompatible with separate process 🙃

Reply:
>  it actually sounds like separate process tenants/sql pods don't support TSDB writes

It does appear that way 🙃. `TenantServer` does export metrics via `tenantRegistry`, and its parent (struct) has corresponding `MetricsRecorder`. However, unlike the shared-process mode, the recorder is not being polled via `GetTimeSeriesData`, so the metrics are not being written to TSDB (via KV). There is slack thread [1], which seems to confirm it, but I can't find an exact issue related to this problem; i.e., metrics from separate-process tenants aren't persisted to TSDB.

@dhartunian Would you have further context on the above?

[1] https://cockroachlabs.slack.com/archives/C06301L45NF/p1729628404945779?thread_ts=1729608426.920709&cid=C06301L45NF

Reply:
Tenant metric persistence in-cluster was not implemented for separate-process multi-tenancy.

I don't think there are any specific blockers to it, but there wasn't a strong need since we don't support separate-process multi-tenancy for self-hosted customers. When we run this in our managed service we use 3rd party metrics persistence solutions which reduce the need for tenant metric persistence. Also in those cases the number of tenants can be quite large and we haven't tested TSDB in those scenarios.

Reply:
Gotcha, thanks for the explanation David!

We don't export tenant metrics to prom either, but it seems more achievable short term than supporting TSDB. I think once we do, using `clusterstats` to query prom for `sql.select.count` instead could be a viable workaround. I added it is as a TODO but as of right now I think this test will have to stay disabled on separate process.

Reply:
Darryl, we can link this issue [1] to the TODO and close this one out.

[1] https://github.com/cockroachdb/cockroach/issues/137625